internal/generator/gen.go

```diff
@@ -12,40 +13,63 @@ import (
 	"strings"
 
 	"github.com/iancoleman/strcase"
+	"golang.org/x/exp/maps"
+
 	"github.com/woven-planet/go-zserio/internal/ast"
 	"github.com/woven-planet/go-zserio/internal/model"
 )
 
-type Options struct {
-	RootPackage       string
-	DoNotFormatSource bool
-	OutputPackage     string
+type options struct {
+	rootPackage       string
+	doNotFormatSource bool
+	outputPackage     string
+	outputToStdout    bool
+	EmitSqlSupport    bool
+}
+
+// Option sets a configuration option
+type Option func(opts *options)
 
-	outputToStdout bool
+// DoNotFormatSource indicates generate Go source should not be formatted
+func DoNotFormatSource(opts *options) {
+	opts.doNotFormatSource = true
+}
+
+// EmitSqlSupport tells the generator to produce support for storing enum
+// types in a SQL database.
+func EmitSqlSupport(opt *options) {
+	opt.EmitSqlSupport = true
```

Comment:
nit, the capitalization here should be `SQL` and not `Sql` according to go conventions.

Reply:
Fixed